index.js

```diff
@@ -1,13 +1,5 @@
 (function() {
-  if (typeof window === "undefined") {
-    return console.error("This environment lacks 'window' support.");
-  }
-
-  if (typeof document === "undefined") {
-    return console.error("This environment lacks 'document' support.");
-  }
-
-  var React = window.React || require('react');
+  var React = typeof window !== 'undefined' && window.React || require('react');
```

Comment:
The checks need to be delayed in order to cope with environments like jsdom, where `window`/`document` are not necessarily defined at module load time.

Also, think about server-side rendering. In the original version, these errors are printed as soon as your module is loaded, even if it isn't used. `componentDidMount` doesn't get run when doing `React.renderToString` or `React.renderToStaticMarkup`, so in my version these errors never get shown unless they're actually relevant.